core/modules/xrdsvc/SsiService.cc

```diff
@@ -79,12 +80,17 @@ SsiService::SsiService(XrdSsiLogger* log) {
         throw wconfig::ConfigError("Couldn't setup scratch db");
     }
 
+    // TODO: set poolSize and all maxThreads values from config file.
+    uint poolSize = std::max(static_cast<uint>(24), std::thread::hardware_concurrency());
+    //_foreman = wcontrol::Foreman::newForeman(std::make_shared<wsched::FifoScheduler>(), poolSize);
+    // TODO: set GroupScheduler group size from configuration file
+    //_foreman = wcontrol::Foreman::newForeman(std::make_shared<wsched::GroupScheduler>(12), poolSize);
+    // poolSize should be greater than either GroupScheduler::maxThreads or ScanScheduler::maxThreads
     _foreman = wcontrol::Foreman::newForeman(
-        std::make_shared<wsched::BlendScheduler>(
-            std::make_shared<wsched::GroupScheduler>(),
-            std::make_shared<wsched::ScanScheduler>()
-        )
-    );
+            std::make_shared<wsched::BlendScheduler>(
+                std::make_shared<wsched::GroupScheduler>(20, 10),
+                std::make_shared<wsched::ScanScheduler>(20)),
+            poolSize);
```

Comment:
Something to think about here is that, as opposed to just controlling how many threads are running, we can easily change the group and scan schedulers to limit how many chunks are being accessed at least in the sense of Tasks being associated with a chunk. The scan scheduler does one chunk at a time. The group scheduler tries to group up Tasks on the same chunk. It would be pretty easy to change the group scheduler to only have X chunks open at a time.